src/components/DragAndDropFile.tsx

```diff
@@ -0,0 +1,106 @@
+        dragging: false,
+    };
+
+    dropRef: any;
+    fileInput: any;
+    dragCounter: number;
+
+    constructor(props: DragAndDropFileProps) {
+        super(props);
+        this.dropRef = React.createRef();
+        this.dragCounter = 0;
+
+        this.handleFileDrag = this.handleFileDrag.bind(this);
+        this.handleFileDragIn = this.handleFileDragIn.bind(this);
+        this.handleFileDragOut = this.handleFileDragOut.bind(this);
+        this.handleFileDrop = this.handleFileDrop.bind(this);
+    }
+
+    componentDidMount(): void {
+        this.dropRef.current.addEventListener('dragenter', this.handleFileDragIn);
+        this.dropRef.current.addEventListener('dragleave', this.handleFileDragOut);
+        this.dropRef.current.addEventListener('dragover', this.handleFileDrag);
+        this.dropRef.current.addEventListener('drop', this.handleFileDrop);
+    }
+
+    componentWillUnmount(): void {
+        this.dropRef.current.removeEventListener('dragenter', this.handleFileDragIn);
+        this.dropRef.current.removeEventListener('dragleave', this.handleFileDragOut);
+        this.dropRef.current.removeEventListener('dragover', this.handleFileDrag);
+        this.dropRef.current.removeEventListener('drop', this.handleFileDrop);
+    }
+
+    handleFileInputChange(event: ChangeEvent<HTMLInputElement>): void {
+        const file = event.target.files ? event.target.files[0] : null;
+        this.props.onFileInputChange(file);
+        console.log('handle file input change');
+    }
+
+    handleFileDragIn(event: DragEvent<HTMLDivElement>): void {
+        event.preventDefault();
+        event.stopPropagation();
+        this.dragCounter++;
+        if (event.dataTransfer.items && event.dataTransfer.items.length > 0) {
+            this.setState({ dragging: true });
+        }
+    }
+
+    handleFileDragOut(event: DragEvent<HTMLDivElement>): void {
+        event.preventDefault();
+        event.stopPropagation();
+        this.dragCounter--;
+        if (this.dragCounter <= 0) {
+            this.setState({ dragging: false });
+        }
+    }
+
+    handleFileDrag(event: DragEvent<HTMLDivElement>): void {
+        event.preventDefault();
+        event.stopPropagation();
+    }
+
+    handleFileDrop(event: DragEvent<HTMLDivElement>): void {
+        event.preventDefault();
+        event.stopPropagation();
+        if (event.dataTransfer.files && event.dataTransfer.files[0]) {
+            this.props.onFileInputChange(event.dataTransfer.files[0]);
+        }
+        this.setState({ dragging: false });
+    }
+
+    render(): React.ReactNode {
+        return (
+            <div>
+                Hello world, this is Drag and Drop component.
+                <div className={`${this.state.dragging && 'file-transfer-container-drag'}`} ref={this.dropRef}>
+                    {this.state.dragging ? (
+                        <div>
+                            <p className="file-transfer-text">Drop file to send.</p>
```

Comment:
I think we should separate the css too into `DragAndDropFile.css` and possibly change the `className`.